pkg/store/tree.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2019-present Open Networking Foundation.
```

Comment:
wondering if this class can actually live in the utils. I'm thinking of this because there is no actual relevance to the store and I'll keep that "store" package for things related to atomic

Reply:
I think the reason I put it in store is because of the risk of circular dependencies. If I moved it to utils then utils would be dependent on store. And then if I ever needed to use anything from utils inside store I would have a circular dependency. Here I was keeping it close to where it is needed

Reply:
+1 for now, we can move later